cosmo_tester/test_suites/summary/__init__.py

```diff
@@ -0,0 +1,599 @@
+import os
+import time
+
+from cloudify_rest_client.exceptions import CloudifyClientError
+import pytest
+import retrying
+
+from cosmo_tester.framework.fixtures import image_based_manager  # noqa
+
+
+manager = image_based_manager
+
+SUMMARY_SNAPSHOT_PATH = os.path.abspath(
+    os.path.join(
+        os.path.dirname(
+            __file__), '..', '..', 'resources/snapshots/testsummarize.zip'))
+SUPPORTED_FIELDS = {
+    'blueprints': [
+        'tenant_name',
+        'visibility',
+    ],
+    'deployments': [
+        'blueprint_id',
+        'tenant_name',
+        'visibility',
+    ],
+    'executions': [
+        'status',
+        'blueprint_id',
+        'deployment_id',
+        'workflow_id',
+        'tenant_name',
+        'visibility',
+    ],
+    'nodes': [
+        'deployment_id',
+        'tenant_name',
+        'visibility',
+    ],
+    'node_instances': [
+        'deployment_id',
+        'node_id',
+        'state',
+        'host_id',
+        'tenant_name',
+        'visibility',
+    ],
+}
+
+
+@pytest.fixture(scope='module')
+def prepared_manager(manager, cfy, logger):
+    manager.client.snapshots.upload(
+        SUMMARY_SNAPSHOT_PATH,
+        'filtering_test_snapshot',
+    )
+    execution = manager.client.snapshots.restore(
+        'filtering_test_snapshot',
+        restore_certificates=False,
+        force=False,
+    )
+    try:
+        wait_for_execution(
+            manager,
+            execution,
+            logger)
+    except ExecutionFailed:
+        # See any errors
+        cfy.executions.list(['--include-system-workflows'])
+        raise
+    # Wait for any post restore commands to run
+    time.sleep(20)
+    yield manager
+
+
+class ExecutionWaiting(Exception):
+    """
+    raised by `wait_for_execution` if it should be retried
+    """
+    pass
```

Comment:
you can omit the `pass` if it already has a docstring

Reply:
Done.